examples/language/openmoe/benchmark/benchmark_train.py

```diff
@@ -0,0 +1,191 @@
+import colossalai
+import datasets
+import torch
+import transformers
+from colossalai import get_default_parser
+from colossalai.booster import Booster
+from colossalai.booster.plugin import LowLevelZeroPlugin
+from colossalai.cluster import DistCoordinator
+from colossalai.logging import disable_existing_loggers, get_dist_logger
+from colossalai.moe.manager import MOE_MANAGER
+from colossalai.moe.utils import skip_init
+from colossalai.utils import get_current_device
+from model.modeling_openmoe import OpenMoeForCausalLM
+from torch.utils.data import Dataset
+from tqdm import tqdm
+from transformers import Adafactor
+from transformers.models.llama import LlamaConfig
+from utils import SimpleTimer, print_model_numel
+
+
+class RandomDataset(Dataset):
+
+    def __init__(self,
+                 num_samples: int = 1000,
+                 max_length: int = 2048,
+                 vocab_size: int = 32000):
+        self.num_samples = num_samples
+        self.max_length = max_length
+        self.input_ids = torch.randint(0, vocab_size,
+                                       (num_samples, max_length),
+                                       device=get_current_device())
+        self.attention_mask = torch.ones_like(self.input_ids,
+                                              device=get_current_device())
+
+    def __len__(self):
+        return self.num_samples
+
+    def __getitem__(self, idx):
+        return {
+            'input_ids': self.input_ids[idx],
+            'attention_mask': self.attention_mask[idx],
+            'labels': self.input_ids[idx]
+        }
+
+
+def parse_args():
+    parser = get_default_parser()
+    # TODO: add model_name
+    # parser.add_argument("--model_name", type=str, default="base", choices=["base", "8b"],
+    #                     help="Path to pretrained model or model identifier from huggingface.co/models.")
+    parser.add_argument("--num_epoch", type=int, default=1, help="Number of epochs.")
+    parser.add_argument("--batch_size", type=int, default=4, help="Batch size (per dp group) for the training dataloader.")
+    parser.add_argument("--seed", type=int, default=42, help="A seed for reproducible training.")
+    parser.add_argument("--num_samples", type=int, default=1000, help="Number of samples in the dataset.")
+
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+
+    LOSS_CONFIG = {
```

Comment:
read from given config, and add or overwrite some values. so that we can test different models

Reply:
Fixed in d7780b7.